src/Microsoft.Management.Deployment/PackageVersionInfo.cpp

```diff
@@ -138,4 +139,33 @@ namespace winrt::Microsoft::Management::Deployment::implementation
     { 
         return m_packageVersion; 
     }
+    Microsoft::Management::Deployment::CatalogPackageMetadata PackageVersionInfo::GetCatalogPackageMetadata()
+    {
+        auto catalogPackageMetadata = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::CatalogPackageMetadata>>();
+        if (m_packageVersion)
+        {
+            auto manifest = m_packageVersion->GetManifest();
+            catalogPackageMetadata->Initialize(manifest.DefaultLocalization);

```

Comment:
As you have implemented this, there is no way to get the values for the locale as configured for the user without the caller having to determine what that locale is.  Was that intentional?  I don't think we should require the caller to do that work unless there is a real reason to force it (and remember that they won't know the settings values).

This function returning the default locale is not useful; I think you misinterpreted my previous comment.  I was referring to the default localization as a concept, not `DefaultLocalization`.  The one that would be used by default within winget, based on the user/system configuration.

I'm fine with not allowing an empty/null string into the other method, but then this one should be doing the proper determination of what the default locale is and then use it.  That is done with `manifest.ApplyLocale()`.

Reply:
I definitely misunderstood what you meant by `DefaultLocalization` 😅 

I decided to keep both methods but changed the one without a parameter to point to the CurrentLocalization after calling `manifest.ApplyLocale()` as suggested. That way the caller can get the default locale values without needing to determine what that locale is and still have the option to get other locale values if needed.